pkg/types/kustomization.go

```diff
@@ -223,3 +227,21 @@ type ImageTag struct {
 	// If digest is present NewTag value is ignored.
 	Digest string `json:"digest,omitempty" yaml:"digest,omitempty"`
 }
+
+// GeneratorOptions provide options to modify the behavior of SecretGenerator and ConfigMapGenerator
+type GeneratorOptions struct {
+	// Labels add labels to all generated secrets and configmaps.
+	Labels map[string]string `json:"labels,omitempty" yaml:"labels,omitempty"`
+
+	// Annotations add annotations to all generated secrets and configmaps.
+	Annotations map[string]string `json:"annotations,omitempty" yaml:"annotations,omitempty"`
+
+	// TimeoutSeconds specifies the timeout for commands.
+	TimeoutSeconds *int64 `json:"timeoutSeconds,omitempty" yaml:"timeoutSeconds,omitempty"`
+
+	// Shell binary, arguments to use for ConfigMap and Secret generators.
+	Shell []string `json:"shell,omitempty" yaml:"shell,omitempty"`
+
+	// DisableHash if set to true, hash won't be appended to generated secrets and configmaps.
```

Comment:
// DisableNameHash if true disables the default behavior of adding a 
// suffix to the names of generated resources that is a hash of the
// resource contents.